compiler/damlc/tests/src/DA/Test/DamlcMultiPackage.hs

```diff
@@ -0,0 +1,760 @@
+            simpleTwoPackageProject
+        , testCache
+            "Only A is rebuild if its Dar is deleted but it's package-id doesn't change"
+            (["--all"], "")
+            [PackageIdentifier "package-a" "0.0.1", PackageIdentifier "package-b" "0.0.1"] -- First time builds both
+            (const $ removeFile "package-a/.daml/dist/package-a-0.0.1.dar") -- Delete package-a/.daml/dist/package-a-0.0.1.dar
+            (["--all"], "")
+            [PackageIdentifier "package-a" "0.0.1"] -- So second time rebuilds only A, as its package-id hasn't changed so B is not stale
+            simpleTwoPackageProject
+        , testCache
+            "A and B rebuild if A's code is modified and A's dar is deleted"
+            (["--all"], "")
+            [PackageIdentifier "package-a" "0.0.1", PackageIdentifier "package-b" "0.0.1"] -- First time builds both
+            -- Modify package-a/daml/PackageAMain.daml, Delete package-a/.daml/dist/package-a-0.0.1.dar
+            (const $ do
+              appendFile "./package-a/daml/PackageAMain.daml" "\nmyDef = 3"
+              removeFile "package-a/.daml/dist/package-a-0.0.1.dar"
+            )
+            (["--all"], "")
+            [PackageIdentifier "package-a" "0.0.1", PackageIdentifier "package-b" "0.0.1"] -- So second time rebuilds A and B, as B depends on A
+            simpleTwoPackageProject
+        , testCache
+            "B rebuilds is A is manually rebuilt after change"
+            (["--all"], "")
+            [PackageIdentifier "package-a" "0.0.1", PackageIdentifier "package-b" "0.0.1"] -- First time builds both
+            -- Modify package-a/daml/PackageAMain.daml, Delete package-a/.daml/dist/package-a-0.0.1.dar
+            (\manualBuild -> do
+              appendFile "./package-a/daml/PackageAMain.daml" "\nmyDef = 3"
+              manualBuild "./package-a"
+            )
+            (["--all"], "")
+            [PackageIdentifier "package-a" "0.0.1", PackageIdentifier "package-b" "0.0.1"] -- Both have been rebuilt
+            simpleTwoPackageProject
+        , testCache
+            "Top package is always built (A)"
+            (["--all"], "")
+            [PackageIdentifier "package-a" "0.0.1", PackageIdentifier "package-b" "0.0.1"] -- First time builds both
+            (const $ pure ())
+            (["--multi-package-search"], "./package-a")
+            [PackageIdentifier "package-a" "0.0.1"]
+            simpleTwoPackageProject
+        , testCache
+            "Top package is always built (B)"
+            (["--all"], "")
+            [PackageIdentifier "package-a" "0.0.1", PackageIdentifier "package-b" "0.0.1"] -- First time builds both
+            (const $ pure ())
+            (["--multi-package-search"], "./package-b")
+            [PackageIdentifier "package-b" "0.0.1"] -- B is rebuilt but gives same package-id, so A is not rebuilt
```

Comment:
```suggestion
            [PackageIdentifier "package-b" "0.0.1"]
```

I believe this comment belongs on the previous test, as `B` depends on `A`, not the other way around

Reply:
Ah yeah this doesn't look right, good spot